selftests/functional/plugin/spawners/process.py

```diff
@@ -31,3 +33,30 @@ def test_logdir_path(self):
         with open(logfile, "r", encoding="utf-8") as debug_file:
             expected = f"logdir is: {testdir}"
             self.assertIn(expected, debug_file.read())
+
+    @unittest.skipUnless(
```

Comment:
IIUIC this test won't be ever clean up, what about to run it only in CI?

Reply:
Interestingly enough, I see no left over process after the main Avocado process driving the test finishes.  I'll investigate the behavior a bit more, and if needed, make it more assertive and attempt to keep it running "everywhere".

Reply:
@richtja I've done some digging on this, and this is what I found.

1. The `avocado-runner-rogue` process opens a network connection to the socket, where it will send status messages to.

```
connect(3, {sa_family=AF_UNIX, sun_path="/var/tmp/avocado_o9mc5xxn/.status_server.sock"}, 48) = 0
```

2. Then it starts sending status messages, as it should:

```
sendto(3, "{\"status\": \"started\", \"time\": 81"..., 268, 0, NULL, 0 <unfinished ...>
```

3. When the job is over, the status server socket is removed, and the `avocado-runner-rogue` gets an error writing to that socket:

```
sendto(3, "{\"status\": \"running\", \"time\": 81"..., 137, 0, NULL, 0) = -1 EPIPE (Broken pipe)
```

4. That becomes a `SIGPIPE`:

```
--- SIGPIPE {si_signo=SIGPIPE, si_code=SI_USER, si_pid=241586, si_uid=1000} ---
```

5. Then it becomes a `BrokenPipeError`:

```
write(2, "    self.connection.send(data.en"..., 70) = 70
write(2, "BrokenPipeError: [Errno 32] Brok"..., 40) = 40
```

Which causes the `avocado-runner-rogue` process to be terminated.

So, this deserves attention in a number of places.  TBH, Avocado is *not* capturing possible crashes on the runners, and this can be improved.  But, looking at the specific point you raised, there won't be any processes leaked in this test.

Reply:
Interesting, thank you for this investigation. You are right that for this test it works fine and we don't need to do anything here. But even though, can you please open an issue about it ? IMO, we don't need to work on this right now, but it will be useful to have an issue about it so we would be aware about this in the future. Thank you again, this is great investigation.

Reply:
> Interesting, thank you for this investigation. You are right that for this test it works fine and we don't need to do anything here. But even though, can you please open an issue about it ? IMO, we don't need to work on this right now, but it will be useful to have an issue about it so we would be aware about this in the future. Thank you again, this is great investigation.

FYI: https://github.com/avocado-framework/avocado/issues/5794